lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -825,13 +825,22 @@ void ProxyRenderDelegate::_Execute(const MHWRender::MFrameContext& frameContext)
         }
     }
 
-    if (_defaultCollection->GetReprSelector() != reprSelector) {
-        _defaultCollection->SetReprSelector(reprSelector);
-        _taskController->SetCollection(*_defaultCollection);
-    }
-
     // if there are no repr's to update then don't even call sync.
     if (reprSelector != HdReprSelector()) {
+        if (_defaultCollection->GetReprSelector() != reprSelector) {
```

Comment:
With the new point snapping support when we do a selection pass we often end up with an empty reprSelector. If that happens don't save it to _defaultCollection, because doing so trashes the previous reprSelector we had saved, and causes the full scene MarkRprimDirty to run. We want to avoid running that as much as possible because the cost of even the empty sync calls is non-zero.